### DISCUSSION TITLE
Does the `helmfile template` command use concurrency

### DISCUSSION THREAD

**Discussion by rochana-atapattu** [helmfile/helmfile#124] [Questions]
Does the `helmfile template` command use concurrency?
at a quick glance, the code does not use the concurrency value. Am I missing something?

Also are there any limitations for parallel execution in helmfiles listed in a helmfile Especially for template command?

**Comment by mark007** [comment]
When dealing with an ever growing helmfile, with lots of large helm charts under it, the time taken to run a helmfile template can be quite long, eg 1 minute, even with the latest version of helm with its templating performance improvements.

I notice from watching the log output that it runs each underlying helm template command in sequence. Can this be parallelized (whether by default ideally, or behind a new flag if really required) so as to speed up the overall 'helmfile template' performance.

I would hope its not a complex change from the helmfile side, but the benefits would be huge.
